types/syncrepl.pp

```diff
@@ -0,0 +1,46 @@
+# @summary Parameters for database replication consumers
+#
+# @see https://www.openldap.org/doc/admin24/slapdconfig.html#syncrepl
+type Openldap::Syncrepl = Struct[
+  {
+    rid                        => Integer[0, 999],
```

Comment:
@coreone @tparkercbn as users of the module you can help: the doc for `rid` says:

> non-negative and is no more than three decimal digits in length

(hence the `Integer[0, 999]`), but I see a lot of examples with exactly 3 digits in length (e.g. `rid=001`), and am wondering what should be done:

1. Keep this as it is (no leading 0);
2. Change it to a String composed of 3 digits (`Pattern[\A\d{3}\z]`);
3. Keep and Integer here but format it as a String composed of 3 digits when serializing the structure;
4. Something else (and what?)

I would like to avoid doing a major version bump and discover that the interface is not great and require another major version bump to be fixed :wink:

Reply:
It accept integer or string up to 3 digit.

For example on my infrastructure it is rid=01

Reply:
From the docs: https://www.openldap.org/doc/admin23/syncrepl.html

> \<rid\> identifies a consumer replica locally within the consumer server. It is used to relate the cookie to the syncrepl definition in slapd.conf \(5\) which has the matching replica identifier. The \<rid\> must have no more than 3 decimal digits.

So, Integer from 0-999 seems like a reasonable validation for this field.

Reply:
@Valantin @coreone thanks for the feedback!  As "padding" zeros are currently used and seems quite usual from my findings and your feedback, I adjusted the data-type to allow to pass the `rid` as a String, using a pattern to match up to 3 digits, so `1`, `"1"`, `"01"` and `"001"` are all valid (and so is `01` but it is the same as `1` but expressed in octal).